aiven/resource_kafka_connector.go

```diff
@@ -241,3 +255,38 @@ func resourceKafkaConnectorState(d *schema.ResourceData, m interface{}) ([]*sche
 
 	return []*schema.ResourceData{d}, nil
 }
+
+// KafkaConnectorWaiter is used to wait for Aiven Kafka Connector list to be available
+type KafkaConnectorWaiter struct {
+	Client      *aiven.Client
+	Project     string
+	ServiceName string
+}
+
+// RefreshFunc will call the Aiven client and refresh it's state.
+func (w *KafkaConnectorWaiter) RefreshFunc() resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		list, err := w.Client.KafkaConnectors.List(w.Project, w.ServiceName)
+		if err != nil {
```

Comment:
Should this be more specific about which error is waited on? If the response is 404 then presumably the connector just doesn't exist.

Reply:
We won't be waiting when an error is 404, only in case of a different errors